packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -2010,6 +2010,40 @@ class _FocusTrap extends SingleChildRenderObjectWidget {
   }
 }
 
+/// Declares a widget subtree which is part of the provided [focusNode]'s focus area
+/// without attaching focus to that region.
+///
+/// This is used by text field widgets which decorate a smaller editable text area.
+/// This area is conceptually part of the editable text, but not attached to the
+/// focus context. The [FocusTrapArea] is used to inform the framework of this
+/// relationship, so that primary pointer contact inside of this region but above
+/// the editable text focus will not trigger loss of focus.
```

Comment:
You might add a reference/description here for the focus trap and when it comes into play, etc. If I pretend that I just happened upon the API docs page for this widget, it feels like if I wasn't writing my own text widget I wouldn't quite know when to use it, or what it affects.

Since `_FocusTrap` isn't a public class, there's probably not much mention of it in other docs.

Reply:
I could make it public, no need to keep it private really

Reply:
that would require me to document it a bit more, but no harm there

Reply:
The other option would be to make `FocusTrapArea` private, if you don't think anyone will need it outside of the text field (but it seems like they might).

Reply:
probably the cupertino text field needs it as well, as would other like the macos ui package - depending on whether or not they are editable text widget based